packages/account/src/BiconomySmartAccount.ts

```diff
@@ -0,0 +1,173 @@
+  SmartAccountFactory_v100,
+  SmartAccountFactory_v100__factory,
+  SmartAccount_v100__factory,
+} from '@biconomy/common'
+import { BiconomySmartAccountConfig } from './utils/Types'
+import { UserOperation, Transaction } from '@biconomy/core-types'
+import NodeClient from '@biconomy/node-client'
+import INodeClient from '@biconomy/node-client'
+import { IBiconomySmartAccount } from 'interfaces/IBiconomySmartAccount'
+import { SupportedChainsResponse, BalancesResponse, BalancesDto, UsdBalanceResponse, SmartAccountByOwnerDto, SmartAccountsResponse, SCWTransactionResponse } from "@biconomy/node-client"
+
+export class BiconomySmartAccount extends SmartAccount implements IBiconomySmartAccount {
+  private factory: SmartAccountFactory_v100
+  private nodeClient: INodeClient
+
+  constructor(readonly biconomySmartAccountConfig: BiconomySmartAccountConfig) {
+    const { signerOrProvider, rpcUrl, epAddress, factoryAddress, bundlerUrl, paymasterUrl, nodeClientUrl } = biconomySmartAccountConfig
+    super({
+      bundlerUrl,
+      epAddress
+    })
+    this.provider = new JsonRpcProvider(rpcUrl);
+    this.entryPoint = EntryPoint_v100__factory.connect(epAddress, this.provider);
+    this.factory = SmartAccountFactory_v100__factory.connect(
+      factoryAddress,
+      this.provider
+    );
+    this.nodeClient = new NodeClient({ txServiceUrl: nodeClientUrl ?? NODE_CLIENT_URL })
+    try {
+      if (Signer.isSigner(signerOrProvider)) {
+        this.signer = signerOrProvider
+      } else if (Provider.isProvider(signerOrProvider)) {
+        this.signer = signerOrProvider.getSigner()
+      }
+      else {
+        console.error('signer or provider is not valid')
+      }
+    } catch (error) {
+      throw new Error("No signer provided")
+    }
+    console.log(paymasterUrl);
+    if (paymasterUrl)
+      this.paymaster = new BiconomyPaymasterAPI(paymasterUrl)
+  }
+  /**
+   * @description This function will initialise BiconomyAccount class state
+   * @returns Promise<BiconomyAccount>
+   */
+  async init(): Promise<BiconomySmartAccount> {
+    const instance = new BiconomySmartAccount({
+      signerOrProvider: this.biconomySmartAccountConfig.signerOrProvider,
+      rpcUrl: this.biconomySmartAccountConfig.rpcUrl,
+      epAddress: this.biconomySmartAccountConfig.epAddress,
+      factoryAddress: this.biconomySmartAccountConfig.factoryAddress,
+      bundlerUrl: this.biconomySmartAccountConfig.bundlerUrl,
+      paymasterUrl: this.biconomySmartAccountConfig.paymasterUrl
+    }
+    );
+    try {
+      instance.chainId = await this.provider.getNetwork().then((net) => net.chainId)
+      await instance.getInitCode()
+      await instance.entryPoint.callStatic.getSenderAddress(instance.initCode);
+
+      throw new Error("getSenderAddress: unexpected result");
+    } catch (error: any) {
+      const addr = await instance.factory.getAddressForCounterFactualAccount(await instance.signer.getAddress(), ethers.BigNumber.from(0))
```

Comment:
Get the index as optional parameter in init method. 0 should be used if index is not provided.

Reply:
resolved. have also added setter for account Index